client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -336,7 +336,7 @@ class DeltaSharingRestClient(
     val target = getTargetUrl(
       s"/shares/$encodedShareName/schemas/$encodedSchemaName/tables/$encodedTableName/metadata" +
         s"$encodedParams")
-    val response = getNDJson(target)
+    val response = getNDJson(target, requireVersion = true, checkEndStreamActionHeader = false)
```

Comment:
Question: do we set includeendstreamaction for metadata queries?
If do, would it be a problem if some DS server returns endStreamAction for it?

Reply:
makes sense, updated to not set includeendstreamaction for table version and metadata queries.